application/templates/deployment.yaml

```
@@ -20,7 +20,7 @@ metadata:
   name: {{ template "application.name" . }}
   namespace: {{ include "application.namespace" . }}
 spec:
-{{- if .Values.deployment.replicas }}
+{{- if not (quote .Values.deployment.replicas | empty) }}
```

```suggestion
{{- if not (eq .Values.deployment.replicas nil) }}
```
I would rather check vs nil instead of empty.